src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -36,16 +33,20 @@ internal class ExternalErrorDiagnosticUpdateSource : IDiagnosticUpdateSource
         private InprogressState _stateDoNotAccessDirectly = null;
         private ImmutableArray<DiagnosticData> _lastBuiltResult = ImmutableArray<DiagnosticData>.Empty;
 
-        [ImportingConstructor]
         public ExternalErrorDiagnosticUpdateSource(
             VisualStudioWorkspaceImpl workspace,
             IDiagnosticAnalyzerService diagnosticService,
             IDiagnosticUpdateSourceRegistrationService registrationService,
-            IAsynchronousOperationListenerProvider listenerProvider) :
+            IAsynchronousOperationListenerProvider listenerProvider,
+            IKnownUIContextService uiContextService) :
```

Comment:
```suggestion
            IKnownUIContextService uiContextServiceOpt) :
```

(or at least until we have the nullable feature, of course!)